src/services/item/plugins/invitation/index.ts

```diff
@@ -192,6 +210,59 @@ const plugin: FastifyPluginAsync = async (fastify) => {
       reply.status(StatusCodes.NO_CONTENT);
     },
   );
+
+  fastify.post<{ Params: { itemId: string } }>(
+    '/:itemId/enroll',
+    {
+      schema: enroll,
+      preHandler: [isAuthenticated, matchOne(validatedMemberAccountRole)],
+    },
+    async ({ user, params }) => {
+      const member = notUndefined(user?.account);
+      assertIsMember(member);
+      const { itemId } = params;
+
+      return await db.transaction(async (manager) => {
+        const repositories = buildRepositories(manager);
+
+        const item = await itemService.get(
+          member,
+          repositories,
+          itemId,
+          PermissionLevel.Read,
+          false,
+        );
+
+        if (!(await itemLoginService.getByItemPath(repositories, item.path))) {
+          const error = new ItemLoginSchemaNotFound();
+          error.statusCode = StatusCodes.FORBIDDEN;
+          throw error;
+        }
+
+        // Check if the member already has a membership, if so, throw an error
+        let hasPermission = true;
+        try {
+          await validatePermission(repositories, PermissionLevel.Read, member, item);
```

Comment:
Perhaps you want to look for memberships directly? This function will not throw if the item is public but the member has no membership. The question is also, do we want to let users to enroll if the item is public? probably yes.